### PR TITLE
Added fix for setting the spine scene to a cloned node

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: defold-spine/src/gui_node_spine.cpp [hunk 1]
```diff
@@ -307,7 +309,8 @@ bool SetScene(dmGui::HScene scene, dmGui::HNode hnode, dmhash_t spine_scene)
     node->m_HasNextCallback = 1;
     node->m_NextCallback = 0;
 
-    return SetupNode(spine_scene, resource, node, true);
+    bool create_bones = node->m_FindBones ? false : true; // See comment in GuiClone()
+    return SetupNode(spine_scene, resource, node, create_bones);
```
Comment (JCash) [inline defold-spine/src/gui_node_spine.cpp:313]: In short, we'll find the gui nodes the next frame.
The cloning of a node is dealt with by the gui system, and it clones the node and its bones.

File: defold-spine/src/gui_node_spine.cpp [hunk 2]
```diff
@@ -683,6 +686,8 @@ static void* GuiClone(const dmGameSystem::CompGuiNodeContext* ctx, const dmGameS
     // But, since the cloned nodes doesn't have any id's, we can't fetch them via id
     // So, we instead create specific gui node type for the bones, and let them register themselves to this cloned node
     SetupNode(src->m_SpinePath, src->m_SpineScene, dst, false);
+    dst->m_FindBones = 1;
+
```
Comment (JCash) [inline defold-spine/src/gui_node_spine.cpp:690]: Just moving it closer to the documentation

File: main/main.gui_script [hunk 3]
```diff
@@ -120,7 +130,8 @@ function init(self)
 
 	for i, bone_id in ipairs(bone_names) do
 		--print("SCRIPT bone", bone_id)
-		local bone = gui.get_node(bone_id)
+        --local bone = gui.get_node(bone_id)
```
Comment (britzl) [inline main/main.gui_script:133]: Indentation

File: main/main.gui_script [hunk 4]
```diff
@@ -109,6 +109,16 @@ function init(self)
 	print("gui.get_spine_scene", gui.get_spine_scene(self.n))
 
 
+    --self.clone = gui.clone(self.n)
```
Comment (britzl) [inline main/main.gui_script:112]: Remove?

**Review by britzl — APPROVED** [review]
Solution looks good. Indentation and commented code.

**Pull request merged**